src/imagery/i.landsat/i.landsat.import/i.landsat.import.py

```diff
@@ -156,7 +156,25 @@ def _untar(inputdir, untardir):
 
     for scene in scenes_to_untar:
         with tarfile.open(name=scene, mode="r") as tar:
-            tar.extractall(untardir)
+
+            def is_within_directory(directory, target):
+
+                abs_directory = os.path.abspath(directory)
+                abs_target = os.path.abspath(target)
+
+                prefix = os.path.commonprefix([abs_directory, abs_target])
+
+                return prefix == abs_directory
+
+            def safe_extract(tar, path=".", members=None, *, numeric_owner=False):
+                for member in tar.getmembers():
+                    member_path = os.path.join(path, member.name)
+                    if not is_within_directory(path, member_path):
+                        raise Exception("Attempted Path Traversal in Tar File")
```

Comment:
```suggestion
                        raise Exception(_("Attempted path traversal in tar file"))
```

Reply:
I'm not 100% sure, but I would expect some linter tool may complain about the use of Exception. [Exception](https://docs.python.org/3/library/exceptions.html#Exception) is a class to derive from and the does not say anything about its direct usage:

> All built-in, non-system-exiting exceptions are derived from this class. All user-defined exceptions should also be derived from this class.

Maybe the better generic exception (for lack of a better match here) is [RuntimeError](https://docs.python.org/3/library/exceptions.html#RuntimeError):

> Raised when an error is detected that doesn’t fall in any of the other categories. The associated value is a string indicating what precisely went wrong.

Reply:
Another aspect is usage of translatable string (`_("...")`) and raising exception versus calling _fatal_. This is generic function, so exception makes sense, but it is actually just nested in a function which uses fatal, so direct fatal makes sense too. Unfortunately, we don't have a best practice in place which would guide us on when to show pure error message with fatal and when to show full traceback caused by an unhanded exception.

Reply:
@wenzeslaus: So what is the decision? Shall we put `RuntimeError` there or shall we call `fatal`? I suggest that one of us would just commit it there and merge the PR.

Reply:
I’ll include the change, for the better string, but still waiting on the decision for `RuntimeError` or `fatal`.